core/src/main/java/io/questdb/cairo/wal/WalWriter.java

```diff
@@ -529,11 +542,11 @@ private void applyMetadataChangeLog(long structureVersionHi) {
                 TableMetadataChange tableMetadataChange = structureChangeCursor.next();
                 try {
                     tableMetadataChange.apply(walMetadataUpdater, true);
-                } catch (SqlException e) {
+                } catch (CairoException e) {
                     distressed = true;
-                    throw CairoException.critical(0)
+                    throw CairoException.critical(0, e)
```

Comment:
i'm not a fan of including bunch of causes in the exception. The original exception can be re-tagged as critical without copying strings around. Also if context is needed, we can just log at the site of catching. The intent of the exception message is to be logged eventually. We don't have to log all strings at once on the same line.

Reply:
I don't mind adding some logging instead of tracking the cause and logging in one go. Going to do that.

Reply:
Done in fd816d0bbb4c76c56373b8a4d6e1e4df19163b06